### PR TITLE
feat: option to configure partition with env var

### LINKED ISSUE #820: Add support for aws-cn partition
I'm working on support for deploying to aws-cn, the wip-pr is here: https://github.com/mbsoftworks/cdk-pipelines-github/pull/1

It works for my use case (though there might be more required changes for other use cases) but I need some help to remove the hard-coded `aws-cn` and `.amazonaws.com.cn` changes (https://github.com/mbsoftworks/cdk-pipelines-github/pull/1/commits/d2ed0187a543b75d5a70408e7a71139eb44c10e9). Any suggestion on what needs to be done to fix those?

### PR DESCRIPTION
**Pull request by mbergkvist**
Adds support for deployment to aws-cn partition by checking an optional environemt variable, `CDK_AWS_PARTITION`. The partition is currently not available to the pipeline in the `jobForDeploy`, adding the environment variable is a non-breaking change to get the pipeline to deploy to aws-cn.

This has been verified with stacks and nested stacks in multiple stages in aws-cn.

Fixes #820

### CODE REVIEW

**Comment by mbergkvist** [comment]
#868

**Review by kaizencc — CHANGES_REQUESTED** [review]
looks good, can we unit test this just for posterity?

**Comment by mbergkvist** [comment]
> looks good, can we unit test this just for posterity?

Absolutely. I'll see what I can do.

> also, maybe a reference in the README would be helpful. doesn't have to be front and center, i don't imagine that many people in aws-cn are using this :)

Of course. I don't know why I didn't add it in the first place 🤦 
Any suggestion under what section this would belong?

**Comment by mbergkvist** [comment]
@kaizencc I've updated the documentation, I'm not a technical writer though so feel free to update it.

I've also added test suites, I figured it would get good coverage by duplicating the current test suites that are relevant for the changes here, and tweaking them for the aws-cn case.

**Comment by mbergkvist** [comment]
@kaizencc Is this ok to be merged?

File: README.md [hunk 1]
```diff
@@ -644,6 +645,19 @@ const pipeline = new GitHubWorkflow(app, 'Pipeline', {
 app.synth();
 ```
 
+## AWS China partition support
+
+The `CDK_AWS_PARTITION` environment variable can be used to specify the AWS partition for the pipeline.
+If it's specified to `aws-cn`, the assets generated by pipeline will reference the resources in
+`.amazonaws.com.cn` instead of `.amazonaws.com.cn`.
```
Comment (kaizencc) [inline README.md:652]:
```suggestion
`.amazonaws.com.cn` instead of `.amazonaws.com`.
```

**Review by kaizencc — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mbergkvist) [reply README.md]: 🤦‍♂️

**Pull request merged**